src/std/fs.ab

```diff
@@ -112,21 +112,21 @@ pub fun glob(path: Text): [Text]? {
 /// Extract the file detecting from the filename the extension
 /// Supports: bz2, gz, xz, bz2, deb, rar, rpm, tar(gz/xz/bz), zip(war/jar), 7z
 /// Note: Not all the commands supports the output folder path
-pub fun extract(path: Text, target: Text): Bool{
+pub fun extract(path: Text, target: Text): Bool? {
     if file_exist(path) {
         if {
-            contains_multiple(path, [".tar.bz2", ".tbz", ".tbz2"]): trust $tar xvjf "{path}" -C "{target}"$
-            contains_multiple(path, [".tar.gz", ".tgz"]): trust $tar xzf "{path}" -C "{target}"$
-            contains_multiple(path, [".tar.xz", ".txz"]): trust $tar xJf "{path}" -C "{target}"$
-            contains_multiple(path, [".bz2"]): trust $bunzip2 "{path}"$
-            contains_multiple(path, [".deb"]): trust $dpkg-deb -xv "{path}" "{target}"$
-            contains_multiple(path, [".gz"]): trust $gunzip "{path}"$
-            contains_multiple(path, [".rar"]): trust $unrar x "{path}" "{target}"$
-            contains_multiple(path, [".rpm"]): trust $rpm2cpio "{path}"| cpio -idm{{}}$
-            contains_multiple(path, [".tar"]): trust $tar xf "{path}"-C "{target}"$
-            contains_multiple(path, [".xz"]): trust $xz --decompress "{path}"$
-            contains_multiple(path, [".7z"]): trust $7z -y "{path}" -o "{target}"$
-            contains_multiple(path, [".zip", ".war", ".jar"]): trust $unzip "{path}" -d "{target}"$
+            contains_any_regex(path, ["/.tar.bz2$/g", "/.tbz$/g", "/.tbz2$/g"]): $ tar xvjf "{path}" -C "{target}" $?
```

Comment:
In which case, we don't really need a `contains_any_regex` function after all.

Reply:
I was thinking that is better to leave to the user how to build the regular expression. In sed you can use also `#` or other symbols as separatore instead of `/`, also you can use different endings instead of `/g` so as it is a regex solution I was thinking that is better to leave to the user how to do what they prefer.

Reply:
Respectfully, I disagree. But I'll have to spend some time composing my reasoning. Will get back to you this evening.

Reply:
> contains_regex(path, "\(\.tar\.bz2\|\.tbz\|\.tbz2\)$")

@hdwalters I don't think we need to escape the parentheses and the pipe symbol. As they don't modify the way text region is parsed in Amber.

```
contains_regex(path, "(\.tar\.bz2\|\.tbz|\.tbz2)$")
```

This snippet should be good enough, right?

Reply:
@Mte90 how about passing flags as optional arguments?

```
fun contains_regex(value, regex, flags = "g") {
    // ...
}
```

Reply:
can be another option indeed, I will do in that way